server/lease/lessor.go

```diff
@@ -572,6 +572,9 @@ func (le *lessor) Detach(id LeaseID, items []LeaseItem) error {
 
 	l := le.leaseMap[id]
 	if l == nil {
+		for _, it := range items {
+			delete(le.itemMap, it)
```

Comment:
OMG, this code.... aaaa 😵

Reply:
Can we add a unit test that will verify integration between lessor and KV store? Because of tight coupling of those two it doesn't make sense to unit test lessor by itself.

Reply:
We have a circular dependency because lack of hierarchy. Looks like that lessor is part of KVstore, meaning it should be higher in hierarchy. Can you please either:
* Replace `fakeLessor` in KVstore to be real lessor. 
* Add new tests to KVstore that use real lessor.

What we should is testing an invariant that call flows:
* `Grant`, `Attach`, `Revoke` and
* `Grant`, `Attach`, `Detach`

Don't leak any objects, aka `itemSet` is empty.

Reply:
@jmhbnz @lavacat Looks like a place for a nice refactor, move lessor under KVstore to enforce hierarchy. Also I remember old proposal to split lessor API based on whether is called from API or from apply flow.

Reply:
> We have a circular dependency because lack of hierarchy. Looks like that lessor is part of KVstore, meaning it should be higher in hierarchy. Can you please either:
> 
> * Replace `fakeLessor` in KVstore to be real lessor.
> * Add new tests to KVstore that use real lessor.
> 
> What we should is testing an invariant that call flows:
> 
> * `Grant`, `Attach`, `Revoke` and
> * `Grant`, `Attach`, `Detach`
> 
> Don't leak any objects, aka `itemSet` is empty.

thx, i will try it later.